gflib/sprite.c

```diff
@@ -326,179 +327,128 @@ void AnimateSprites(void)
 
 void BuildOamBuffer(void)
 {
-    u8 temp;
-    UpdateOamCoords();
-    BuildSpritePriorities();
-    SortSprites();
-    temp = gMain.oamLoadDisabled;
-    gMain.oamLoadDisabled = TRUE;
-    AddSpritesToOamBuffer();
-    CopyMatricesToOamBuffer();
-    gMain.oamLoadDisabled = temp;
-    sShouldProcessSpriteCopyRequests = TRUE;
-}
+    bool32 oamLoadDisabled;
+    u32 i, stride;
+    u8 oamIndex;
+
+    // All attributes which affect sorting packed into a single u32:
+    // { priority:2, subpriority:8, y:9, :5, index:8 }.
+    // Index has its own byte even though it only needs 6 bits so that
+    // we can load it with a ldrb instead of having to mask out the
+    // bottom 6 bits.
+    u32 spritePriorities[MAX_SPRITES];
+    s32 toSort = 0;
+    u8 skippedSprites[MAX_SPRITES];
+    u32 skippedSpritesN = 0;
+    u32 matrices = 0;
```

Comment:
Hmm... This is really just a comment, and I think its probably fine. Just note that this is quite heavy on the stack here. `BuildOamBuffer` is typically just at the top of the call chain inside the main loop, so I'm not too worried about it.

Reply:
Yeah, I came to the same conclusion as you that while it's heavy on the stack we call `BuildOamBuffer` with a mostly clear stack so it should be safe enough :crossed_fingers: